src/utils/FileDescriptor.cpp

```diff
@@ -0,0 +1,77 @@
+#include <hyprutils/utils/FileDescriptor.hpp>
+#include <fcntl.h>
+#include <sys/poll.h>
+#include <unistd.h>
+#include <utility>
+
+using namespace Hyprutils::Utils;
+
+CFileDescriptor::CFileDescriptor(int fd) : m_fd(fd) {}
+
+CFileDescriptor::CFileDescriptor(CFileDescriptor&& other) : m_fd(std::exchange(other.m_fd, -1)) {}
+
+CFileDescriptor& CFileDescriptor::operator=(CFileDescriptor&& other) {
+    if (this == &other)
+        return *this;
+
+    reset();
+    m_fd = std::exchange(other.m_fd, -1);
+    return *this;
+}
+
+CFileDescriptor::~CFileDescriptor() {
+    reset();
+}
+
+bool CFileDescriptor::isValid() const {
+    return m_fd != -1;
+}
+
+int CFileDescriptor::get() const {
+    return m_fd;
+}
+
+int CFileDescriptor::take() {
+    return std::exchange(m_fd, -1);
+}
+
+void CFileDescriptor::reset() {
+    if (m_fd != -1) {
+        close(m_fd);
+        m_fd = -1;
+    }
+}
+
+CFileDescriptor CFileDescriptor::duplicate() const {
+    if (m_fd == -1)
+        return {};
+
+    return CFileDescriptor{fcntl(m_fd, F_DUPFD_CLOEXEC, 0)};
+}
+
+bool CFileDescriptor::isClosed() const {
+    return isClosed(m_fd);
+}
+
+bool CFileDescriptor::isReadable() const {
+    return isReadable(m_fd);
+}
+
+bool CFileDescriptor::isClosed(int fd) {
```

Comment:
this doesn't do what it says? Closed means we closed it. this checks if it's readable from + eats

why not just return m_fd == -1

why is this a static method? what is this for?